theano/gpuarray/opt.py

```diff
@@ -751,16 +751,12 @@ def local_gpua_elemwise(op, context_name, inputs, outputs):
         gpu_output = res(*new_inputs)
         return [gpu_output]
     elif op.scalar_op in (scalar.add, scalar.mul):
-        max_nb_inputs = max_inputs_to_GpuElemwise(outputs)
-        if max_nb_inputs > 1:
-            while len(inputs) > max_nb_inputs:
-                inputs = inputs[:-max_nb_inputs] + [res(*inputs[-max_nb_inputs:])]
-        return res(*inputs)
+        return split_huge_add_or_mul(outputs[0].owner, res).outputs
```

Comment:
This is how we handle the optimizaton when we move to the GPU in the new back-end. This make the optimization faster. We move the full model at once instead of each node at a time. So much less intermediate work that don't stay in the final graph.

outputs.owner.inputs is the original (on the CPU inputs). Inputs is the new inputs we want to use (probably moved to the GPU). This is why we didn't give node, we don't want node.inputs to be used.